vault/core.go

```diff
@@ -765,6 +766,27 @@ func (c *Core) fetchACLTokenEntryAndEntity(clientToken string) (*ACL, *TokenEntr
 		return nil, nil, nil, logical.ErrPermissionDenied
 	}
 
+	// CIDR checks bind all tokens except non-expiring root tokens
+	if te.TTL != 0 && len(te.BoundCIDRs) > 0 {
+		var valid bool
+		remoteSockAddr, err := sockaddr.NewSockAddr(req.Connection.RemoteAddr)
+		if err != nil {
+			if c.Logger().IsDebug() {
+				c.Logger().Debug("could not parse remote addr into sockaddr", "error", err, "remote_addr", req.Connection.RemoteAddr)
+			}
+			return nil, nil, nil, logical.ErrPermissionDenied
+		}
+		for _, cidr := range te.BoundCIDRs {
+			if cidr.Contains(remoteSockAddr) {
+				valid = true
+				break
+			}
+		}
+		if !valid {
```

Comment:
Can we add a server log entry here indicating the reason for failure? It otherwise may be hard to find out the cause for the denial. Or, how about a new error to indicate that this is from a CIDR check failure?

Reply:
We don't do that for any other type of authorization denied. If we're going to start naming types (which I'm not convinced we should do) then we should do it throughout.